File: wk_6_python/probset_6/sentimental-readability/readability.py
def main():
    text = input("Text: ")

    letters, words, sentences = count_letters_words_sentences(text)  # counting all the letters, words, sentences
    L = (letters / words) * 100

    S = (sentences / words) * 100

    index = 0.0588 * L - 0.296 * S - 15.8  # Substituting values into formula
    index = round(index)

    if index < 1:
        print("Before Grade 1")
    elif 1 <= index <= 15:
        print(f"Grade {index}")
    else:
        print("Grade 16+")


def count_letters_words_sentences(text):
    text = text.lower()
    letters = 0
    words = 1  # last word doesn't have any space so intialising words at 1
    sentences = 0

    for letter in text:
        if letter.isalpha():   # counting each letter
            letters += 1
        if letter.isspace():   # counting each word as every word has space after
            words += 1
        if letter in [".", "?", "!"]:  # counting sentences as every sentence ends with '.' or "?" or "!"
            sentences += 1
    return letters, words, sentences


main()

File: wk_2_arrays/probset_2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);


int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text); // getting number of letters
    int words = count_words(text);    // getting number of words
    float L = ((float)letters / words) * 100;

    int sentences = count_sentences(text);    // getting number of sentences
    float S = ((float)sentences / words) * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8; // applying the formula
    int round_index = round(index);

    if (round_index < 1)
    {
        printf("Before Grade 1\n"); // printing the grades
    }
    else if (round_index <= 15 && round_index >= 1)
    {
        printf("Grade %i\n", round_index);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isupper(text[i])) // everytime isupper encounters a aplabet it adds to letters
        {
            letters++;
        }
        if (islower(text[i])) // everytime islower encounters a aplabet it adds to letters
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
// after every word there is space except for last word os the text, so we add 1 word for each space and initialize it at 1 to compansate for last word.
    int words = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?') // if text ends with `.` or `!` or `?` we add a sentence
        {
            sentences++;
        }
    }
    return sentences;
}

